app/src/main/kotlin/com/wire/android/ui/home/AppSyncViewModel.kt

```diff
@@ -77,6 +84,9 @@ class AppSyncViewModel @Inject constructor(
                 viewModelScope.launch { featureFlagsSyncWorker.execute() },
                 viewModelScope.launch { observeCertificateRevocationForSelfClient.invoke() },
                 viewModelScope.launch { updateApiVersions() },
+                viewModelScope.launch { mLSClientManager() },
+                viewModelScope.launch { mLSMigrationManager() },
+                viewModelScope.launch { keyingMaterialsManager() },
```

Comment:
Instead of having the app explicitly knowing about MLS and how it works, can't we just mutate a `hasForegroundPriorityState` thing, and let the business logic handle itself in a single place within `Kalium`? 

Every time we add a new guy, we need to remember to do it here too. Also, other tools like TestService or whatever might be completely broken with these changes.

It would be similar to `NotifyConversationisOpenUseCase`, instead of having the ViewModels calling multiple UseCases, we centralise operations in a single point (clean up ephemeral, resolve 1:1, etc).

Reply:
that will be a nice solution for a bigger refactor, there are quite a lot of those worker that runs on kalium init that i would like to make em all cleaner, but this will take quite some time that we do not have, for now what do you think about this approach that we can revisit later?

Reply:
for me the most important part is to not run all of this on every sync live update

Reply:
Just... other than increasing debt, it might still break TestService (not sure), so QA might not be the happiest about this "temporary change".